### DISCUSSION TITLE
Is  SQLite3MultipleCiphers FIPS compliant?

### DISCUSSION THREAD

**Discussion by chandrakanth-amuru** [utelle/SQLite3MultipleCiphers#69] [Q&A] [answered]
Hi All,
I'm working for a tool that is FIPS 140-2 compliant and we exploring if SQLite3MultipleCiphers  can be used for our project, as part of it would like to know if SQLite3MultipleCiphers is FIPS compliant. Could someone clarify?
Thanks.

**Comment by utelle** [comment, marked as answer]
_SQLite3MultipleCiphers_ is **not** _FIPS 140-2_ validated. In theory, _SQLite3MultipleCiphers_ could be made FIPS compliant by replacing the current implementation of cryptographic algorithms for example by calls to the [OpenSSL](https://github.com/openssl) FIPS module. However, at the moment I have no plans to implement selectable cryptographic providers for _SQLite3MultipleCiphers_. 

There exist at least 2 FIPS-compliant encryption extensions for SQLite:

1. The creators of SQLite offer a commercial [SQLite Encryption Extension](https://www.sqlite.org/see) that can be linked with OpenSSL.
2. The company _Zetetic_, creator of [SQLCipher](https://www.zetetic.net/sqlcipher/), offers a commercial [FIPS-compliant SQLCipher variant](https://www.zetetic.net/sqlcipher/fips/).

**Reply by chandrakanth-amuru** [reply]
Thank you for the info.
